open-weather-connector-demo/src/com/axonivy/connector/openweather/managedbean/ForecastWeatherBean.java

```diff
@@ -0,0 +1,431 @@
+		this.currentChartWindowEndX = currentChartWindowEndX;
+	}
+
+	public int prepareCurrentChartWindowStartX() {
+		return dailyForecastDisplayInfos.get(selectedDateIndex).getChartWindowStartX();
+	}
+
+	public int prepareCurrentChartWindowEndX() {
+		return dailyForecastDisplayInfos.get(selectedDateIndex).getChartWindowEndX();
+	}
+
+	public void search() {
+		processAndGroupForecastData();
+		setSelectedDateIndex(0);
+		createTemperatureModel();
+		createPrecipitationModel();
+	}
+
+	private void processAndGroupForecastData() {
+		Optional<Forecast> optionalForecast = ForecastService.getInstance()
+				.fetchForecastThreeHourlyFiveDay(searchCityName, searchCountryCode, searchStateCode, units);
+
+		if (optionalForecast.isEmpty()) {
+			return;
+		}
+		Forecast forecast = optionalForecast.get();
+
+		forecast.getList().forEach(record -> {
+			record.getWeather().forEach(weather -> {
+				String icon = weather.getIcon();
+				if (icon != null) {
+					icon = icon.replace('n', 'd');
+					weather.setIcon(icon);
+				}
+			});
+		});
+
+		zoneId = ZoneId.ofOffset("UTC", ZoneOffset.ofTotalSeconds(forecast.getCity().getTimezone()));
+		cityName = forecast.getCity().getName();
+		threeHourlyFiveDayForecasts = forecast.getList();
+		groupForecastData();
+		temperatureData = prepareTemperatureData().toString();
+	}
+
+	private void groupForecastData() {
+		Map<LocalDate, List<WeatherRecord>> forecastsByDate = groupForecastsByDate();
+
+		dailyForecasts = generateDailyForecasts(forecastsByDate);
+
+		dailyForecastDisplayInfos = IntStream.range(0, dailyForecasts.size())
+				.mapToObj(index -> createDailyForecastDisplayInfo(index)).collect(Collectors.toList());
+	}
+
+	private Map<LocalDate, List<WeatherRecord>> groupForecastsByDate() {
+		return threeHourlyFiveDayForecasts.stream().collect(
+				Collectors.groupingBy(record -> Instant.ofEpochSecond(record.getDt()).atZone(zoneId).toLocalDate()));
+	}
+
+	private List<DailyForecast> generateDailyForecasts(Map<LocalDate, List<WeatherRecord>> forecastsByDate) {
+		return forecastsByDate.entrySet().stream().map(entry -> new DailyForecast(entry.getKey(), entry.getValue()))
+				.sorted(Comparator.comparing(DailyForecast::getDate)).limit(5).collect(Collectors.toList());
+	}
+
+	private DailyForecastDisplayInfo createDailyForecastDisplayInfo(int index) {
+		int startX = calculateModelStartX(index);
+		int endX = startX + chartWindowSize - 1;
+		return new DailyForecastDisplayInfo(dailyForecasts.get(index), startX, endX);
+	}
+
+	private int calculateModelStartX(int index) {
+		return IntStream.range(0, index).map(i -> dailyForecasts.get(i).getDailyRecords().size()).sum();
+	}
+
+	private void createTemperatureModel() {
+		temperatureModel = new LineChartModel();
+		LineChartOptions options = new LineChartOptions();
+		temperatureModel.setOptions(options);
+		temperatureModel.setData(prepareTemperatureChartData());
+		temperatureModel.setExtender("temperatureChartExtender");
```

Comment:
Create static final String for "temperatureChartExtender". Check others string also